src/lib/hooks/use-safe-client.ts

```diff
@@ -0,0 +1,115 @@
+import SafeApiKit from '@safe-global/api-kit'
+import { EIP712TypedData as LegacyEIP712TypedData } from '@safe-global/api-kit/dist/src/types/safeTransactionServiceTypes'
+import Safe, {
+  buildSignatureBytes,
+  EthSafeSignature,
+  hashSafeMessage,
+  SigningMethod,
+} from '@safe-global/protocol-kit'
+import { EIP712TypedData } from '@safe-global/safe-core-sdk-types'
+import { useEffect, useMemo, useState } from 'react'
+import { useConnectorClient } from 'wagmi'
+
+import { useWallet } from '@/lib/hooks/use-wallet'
+
+export function useSafeClient() {
+  const { address } = useWallet()
+  const [protocolKit, setProtocolKit] = useState<Safe | null>(null)
+  const [safeAddress, setSafeAddress] = useState<string | null>(null)
+  const [safes, setSafes] = useState<string[] | null>(null)
+  const apiKit = useMemo(() => new SafeApiKit({ chainId: BigInt(1) }), [])
+  const { data: connectorClient } = useConnectorClient()
+
+  useEffect(() => {
+    async function loadProtocolKit() {
+      if (!address || !safeAddress || !connectorClient) return
+
+      const protocolKit = await Safe.init({
+        provider: connectorClient.transport,
+        safeAddress,
+        signer: address,
+      })
```

Comment:
You might wanna add a try/catch here and not change buttons if there is a problem initializing

Reply:
Added 👍